rs/cycles_account_manager/src/lib.rs

```diff
@@ -460,6 +461,32 @@ impl CyclesAccountManager {
         self.consume_with_threshold(system_state, cycles, threshold, use_case, reveal_top_up)
     }
 
+    /// Withdraws and consumes the cost of executing the given number of
+    /// instructions.
+    pub fn consume_instructions(
```

Comment:
nit: I'd name this `consume_cycles_for_instructions` instead. `consume_instructions` might give the idea that we're charging instructions whereas we always charge cycles (and it's for instructions executed in this case).